hicommonwealth/commonwealth
# `[Launchpad Discovery]`: New community filters and sorting options

## Link to Issue
Closes: https://github.com/hicommonwealth/commonwealth/issues/8801

## Description of Changes
- Added support for new filters on the communities explore page.
- New filters changes include
    - Added `Sort by` => possible values => `Most Recent`, `Market Cap`, `Price`, `Number of Members`, `Number of Threads`. ATM `Market Cap` and `Price` filters won't work. They will be done in https://github.com/hicommonwealth/commonwealth/issues/9694
    - Added `Sort order` => possible values => `Ascending`, `Descending`
    - Added `Community type` => possible values => `Launchpad`, `Basic`
        - when launchpad is selected it will only show communities associated to a token
        - when basic is selected it will only show communities that are not associated to any token
        - when this filter is removed it will show all communities
   - Updated `Community Ecosystem` => 3 new filters additions => `Base`, `Blast`, `Polygon` - when any of these are selected it will filter communities by the correct eth/cosmos chain id
   - Updated `Community Ecosystem` => 2 filter removals => `Substrate` and `NEAR`
- Some filter UI elements / updated labels are not behind the feature flag, this is intentional.

## "How We Fixed It"
N/A

## Test Plan
1. Enable `FLAG_TOKENIZED_COMMUNITY`
2. Visit the `/communities` page
3. Verify you see updated layout for filters section (I didn't add this behind feature flag intentionally, this is some UI only change that doesn't introduce new features)
4. Verify you are able to see new filter changes in the filters drawer - these should only be visible when `FLAG_TOKENIZED_COMMUNITY` is enabled
5. Verify that each filter you apply works as described in the description section above.
6. Verify that you can combine multiple filters to narrow down results ex: `sort "Basic" community type on "Base" ecosystem, that have self identified with "DAO" preference and "Order by" Number of members with most members first (Order direction = descending)`

## Deployment Plan
N/A

## Other Considerations
N/A
